src/qutip_tensorflow/core/data/convert.py

```diff
@@ -1,33 +1,62 @@
 import qutip
-from .tftensor import TfTensor
+from .tftensor import TfTensor128, TfTensor64
 import tensorflow as tf
 
 # Conversion function
-def _tf_from_dense(dense):
-    return TfTensor(dense.to_array())
+def _tf128_from_dense(dense):
+    return TfTensor128(dense.to_array())
+
+
+def _tf64_from_dense(dense):
+    return TfTensor64(dense.to_array())
 
 
 def _tf_to_dense(tftensor):
     return qutip.data.Dense(tftensor.to_array(), copy=False)
 
 
-def is_tftensor(data):
-    return isinstance(data, tf.Tensor)
+def _tf64_from_tf128(tftensor):
+    # Posibly raise a warning?
```

Comment:
I guess this is only problematic if it happens accidentally (e.g. if the dispatcher chooses to use it for some reason).